nowriter/tmvarint/tm_varint_v0.go

```diff
@@ -0,0 +1,90 @@
+package tmvarint
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"github.com/tendermint/go-wire/nowriter/tmlegacy"
+)
+
+type TMVarintV0 struct {
+}
+
+var _ TMVarint = (*TMVarintV0)(nil)
+var legacy = tmlegacy.TMEncoderLegacy{}
+
+func (e TMVarintV0) EncodeUvarint(i uint) []byte {
+	var b bytes.Buffer
+	w := bufio.NewWriter(&b)
+	var inst_n int
+	n := &inst_n
+	var inst_err error
+	err := &inst_err
+
+	var size = uvarintSize(uint64(i))
+	legacy.WriteUint8(uint8(size), w, n, err)
+	if size > 0 {
+		var buf [8]byte
+		binary.BigEndian.PutUint64(buf[:], uint64(i))
+		legacy.WriteTo(buf[(8-size):], w, n, err)
```

Comment:
what if size > 8? or it can't be?

Reply:
don't know assumption -- based on old decoding code

Reply:
weird. maybe @ebuchman knows?

Reply:
look at implementation of uvarintSize() haha.  it can't be > 8.